### PR TITLE
fix: ambiguity in route2 definitions.

### LINKED ISSUE #8729: cy.route2() doesn't work when method is used as the first parameter
### Current behavior
When passing `POST` as the first parameter to `route2` requests are not properly intercepted.

```js
// Does not work
cy.route2('POST', `${BasePage.api_url}/events/createTemplate`).as('create_template')
cy.wait('@create_template')

// Works
cy.route2(`${BasePage.api_url}/events/createTemplate`).as('create_template')
cy.wait('@create_template')
```

### Desired behavior
Using the method parameter should work.

### Test code to reproduce
https://github.com/bhgsbatista/cypress-test-tiny/tree/route2_method_fail

### Versions
Cypress v5.3.0
Windows 10

### PR DESCRIPTION
**Pull request by sainthkh**
- Closes #8729 

### User facing changelog

There was an ambiguity between route2() definitions. `cy.route2(string, string)` is only resolved as `cy.route2(url: string, routeHandler: string)`. This PR solves this problem.

### Additional details

- Why was this change necessary?  => route2() doesn't work as doc says.
- What is affected by this change? => N/A
- Any implementation details to explain? => It checks the first argument is a value of HTTP or WebDAV request methods.

### How has the user experience changed?

N/A

### PR Tasks

- [x] Have tests been added/updated?
- [x] Has the original issue been tagged with a release in ZenHub? 
- [N/A] Has a PR for user-facing changes been opened in [`cypress-documentation`](https://github.com/cypress-io/cypress-documentation)? 
- [x] Have API changes been updated in the [`type definitions`](https://github.com/cypress-io/cypress/blob/develop/cli/types/cypress.d.ts)?
- [N/A] Have new configuration options been added to the [`cypress.schema.json`](https://github.com/cypress-io/cypress/blob/develop/cli/schema/cypress.schema.json)?

### CODE REVIEW

**Comment by sainthkh** [comment]
Is the failed npm-webpack test a flaky test?

**Comment by sainthkh** [comment]
Flaky failures.

File: packages/driver/src/cy/net-stubbing/add-command.ts [hunk 1]
```diff
@@ -91,6 +91,13 @@ function isNumberMatcher (obj): obj is NumberMatcher {
   return Array.isArray(obj) ? _.every(obj, _.isNumber) : _.isNumber(obj)
 }
 
+function isMethod (arg: string) {
+  const httpRequestMethods = ['get', 'head', 'post', 'put', 'delete', 'connect', 'options', 'trace', 'patch']
```
Comment (jennifer-shehane) [inline packages/driver/src/cy/net-stubbing/add-command.ts:95]: We have a util in the driver already to check for this: https://github.com/cypress-io/cypress/blob/develop/packages/driver/src/cypress/utils.js#L277:L277

Uses the `method` node module.

File: packages/driver/cypress/integration/commands/net_stubbing_spec.ts [hunk 2]
```diff
@@ -172,6 +172,21 @@ describe('network stubbing', function () {
...
+
+      cy.window().then((win) => {
+        win.eval(
+          `fetch("http://dummy.restapiexample.com/api/v1/create", {
+            method: 'POST', // *GET, POST, PUT, DELETE, etc.
+          });`,
+        )
+      })
+
+      cy.wait('@create') // wait times out
```
Comment (jennifer-shehane) [inline packages/driver/cypress/integration/commands/net_stubbing_spec.ts:187]: Remove comment about timing out
```suggestion
      cy.wait('@create')
```

**Review by jennifer-shehane — DISMISSED** [review]
Manually tested that this fixes the original issue.

File: packages/driver/src/cy/net-stubbing/add-command.ts (hunk 1 above)
Comment (sainthkh) [reply packages/driver/src/cy/net-stubbing/add-command.ts]: Done.

File: packages/driver/cypress/integration/commands/net_stubbing_spec.ts (hunk 2 above)
Comment (sainthkh) [reply packages/driver/cypress/integration/commands/net_stubbing_spec.ts]: Deleted.

File: packages/net-stubbing/lib/external-types.ts [hunk 3]
```diff
@@ -176,8 +244,8 @@ export interface RouteMatcherOptionsGeneric<S> extends RouteMatcherCompatOptions
    */
   https?: boolean
   /**
-   * Match against the request's HTTP method.
-   * @default 'GET'
+   * Match against the request's HTTP or WebDAV method.
```
Comment (flotwig) [inline packages/net-stubbing/lib/external-types.ts:247]:
```suggestion
   * Match against the request's HTTP method.
```

Let's keep it simple, 99% of users won't be using WebDAV. Plus, I kinda feel like if it's in the HTTP header's method field, it's an HTTP method, so :shrug:

**Review by flotwig — APPROVED** [review]
Lovely! Thanks

**Pull request merged**